File: Python/Leetcode 45.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        Required_Jumps = [0] * len(nums)
        # Required_Jumps[0] = 1
        # print(Required_Jumps)

        for index in range(len(nums) - 2,-1,-1):
            
            max_jumps = nums[index]
            temp_jumps = 1000000000

            for jump in range(1,max_jumps+1,1):
                if index + jump < len(nums):
                    temp_jumps = min(temp_jumps,Required_Jumps[index + jump])
            
            Required_Jumps[index] = temp_jumps + 1

        # print(Required_Jumps)
        return Required_Jumps[0]

File: C++/Leetcode 45.cpp
#include <stdio.h>
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int jump(vector<int>& nums) {
        vector<int> required_jumps(nums.size(),0);

        for (int i = nums.size() - 2; i >= 0; i--)
        {
            int total_jumps = nums[i];
            int temp_jumps_required = INT_MAX-1;

            for (int j = 1; j <= total_jumps && i + j < nums.size(); j++)
            {
                temp_jumps_required = min(temp_jumps_required, required_jumps[i+j]);
            }
            required_jumps[i] = temp_jumps_required + 1;
        }
        return required_jumps[0];
    }
};